config-vault/src/main/java/io/scalecube/config/vault/VaultConfigSource.java

```diff
@@ -0,0 +1,111 @@
+package io.scalecube.config.vault;
+
+import io.scalecube.config.ConfigProperty;
+import io.scalecube.config.ConfigSourceNotAvailableException;
+import io.scalecube.config.source.ConfigSource;
+import io.scalecube.config.source.LoadedConfigProperty;
+import io.scalecube.config.source.LoadedConfigProperty.Builder;
+
+import com.bettercloud.vault.EnvironmentLoader;
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import com.bettercloud.vault.VaultException;
+import com.bettercloud.vault.response.LogicalResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a {@link ConfigSource} implemented for Vault
+ * 
+ * @author AharonHa
+ * @see The Vault Project's site at {@link https://www.vaultproject.io/}
+ *
+ */
+public class VaultConfigSource implements ConfigSource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(VaultConfigSource.class);
+  private Vault vault;
+  private String SECRET_DEFAULT_PATH;
+
+  /**
+   * This constructor is used internally for test purposes. please use it only for tests
+   * 
+   * @param environmentLoader an {@link EnvironmentLoader}
+   */
+  VaultConfigSource(EnvironmentLoader environmentLoader) {
+    this(Optional.of(new VaultConfig().environmentLoader(environmentLoader)));
+    SECRET_DEFAULT_PATH = environmentLoader.loadVariable("VAULT_SECRETS_PATH");
+  }
+
+  public VaultConfigSource() {
+    this(Optional.empty());
+  }
+
+  /**
+   * Create a new {@link VaultConfigSource} with the given {@link VaultConfig}. <br>
+   * Default configurations can also be used by passing {@link Optional#empty() empty}. Please note the following
+   * required environment variables are required if the configuration does not provide them
+   * <ul>
+   * <li><code>VAULT_SECRETS_PATH</pre> is the path to use (defaults to <code>secret</code>)</li>
+   * <li><code>VAULT_TOKEN</code> is the {@link VaultConfig#token(String) token} to use</li>
+   * <li><code>VAULT_ADDR</code> is the {@link VaultConfig#address(String) address} of the vault (API)</li>
+   * </ul>
+   * 
+   * @param config an optional configuration to create vault access with.
+   * 
+   */
+
+  public VaultConfigSource(Optional<VaultConfig> config) {
```

Comment:
Looks like we need only 
1. token
2. address
3. and VAULT_SECRETS_PATH

Reply:
+new builder class